### PR TITLE
loongarch: various architecture specific fixes.

### PR DESCRIPTION
**Pull request by yxd-ym**
Adopt some patches from https://github.com/yxd-ym/zig-bootstrap/pull/1 .

### CODE REVIEW

File: lib/std/Target.zig [hunk 1]
```diff
@@ -1140,6 +1140,13 @@ pub const Cpu = struct {
             };
         }
 
+        pub inline fn isLoongArch(arch: Arch) bool {
```
Comment (alexrp) [inline lib/std/Target.zig:1143]: This conflicts with my #20894; would you mind leaving this part out?

File: lib/std/mem.zig [hunk 2]
```diff
@@ -17,6 +17,14 @@ pub const page_size = switch (builtin.cpu.arch) {
         else => 4 * 1024,
     },
     .sparc64 => 8 * 1024,
+    .loongarch32, .loongarch64 => switch (builtin.os.tag) {
```
Comment (alexrp) [inline lib/std/mem.zig:20]: Just noting that #20511 will overhaul this, but I think it's ok to merge this change for now.

**Review by alexrp — APPROVED** [review]

File: lib/std/Target.zig (hunk 1 above)
Comment (yxd-ym) [reply lib/std/Target.zig]: Removed.

**Comment by andrewrk** [comment]
Thanks!

**Pull request merged**